src/CsvWriter/MappingCsvWriter.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\AzureStorageTableExtractor\CsvWriter;
+
+use Keboola\AzureStorageTableExtractor\Configuration\Config;
+use Keboola\AzureStorageTableExtractor\Exception\ApplicationException;
+use Keboola\AzureStorageTableExtractor\Exception\UserException;
+use Keboola\AzureStorageTableExtractor\IncrementalFetchingHelper;
+use Keboola\Component\JsonHelper;
+use Keboola\CsvMap\Exception\CsvMapperException;
+use Keboola\CsvMap\Mapper;
+use Keboola\CsvTable\Table;
+
+class MappingCsvWriter extends BaseCsvWriter implements ICsvWriter
+{
+    private Mapper $mapper;
+
+    public function __construct(string $dataDir, Config $config, IncrementalFetchingHelper $incFetchingHelper)
+    {
+        parent::__construct($dataDir, $config, $incFetchingHelper);
+        try {
+            $this->mapper = new Mapper($this->config->getMapping(), false, $this->config->getOutput());
+        } catch (CsvMapperException $e) {
+            throw new UserException($e->getMessage(), $e->getCode(), $e);
+        }
+    }
+
+    public function writeItem(object $item): void
+    {
+        parent::writeItem($item);
+        $this->unsetODataMetadata($item);
+
+        // Ensure UNIQUE FK for sub-documents with the SAME CONTENT, but from the DIFFERENT parent document
+        $userData = ['parentId' => md5(serialize($item))];
+        try {
+            $this->mapper->parseRow($item, $userData);
+        } catch (CsvMapperException $e) {
+            throw new UserException($e->getMessage(), $e->getCode(), $e);
+        }
+    }
+
+    public function finalize(): void
+    {
+        $this->copyTempCsvFiles();
+        $this->writeManifest();
+    }
+
+    protected function copyTempCsvFiles(): void
+    {
+        foreach ($this->mapper->getCsvFiles() as $csvTable) {
+            /** @var Table|null $csvTable */
+            if (!$csvTable) {
+                // Skip, no row
+                continue;
+            }
+
+            // Check file size
+            $source = $csvTable->getPathName();
+            $dest = $this->getCsvTargetPath($csvTable);
+            $filesize = filesize($source);
+            if ($filesize === false) {
+                throw new ApplicationException(sprintf('Failed to get file size "%s".', $source));
+            } elseif ($filesize === 0) {
+                // No rows -> no CSV file
+                continue;
+            }
+
+            // Copy
+            $result = copy($source, $dest);
+            if (!$result) {
+                throw new ApplicationException(sprintf('Failed to copy "%s" -> "%s".', $source, $dest));
+            }
+        }
+    }
+
+    protected function writeManifest(): void
```

Comment:
pro ukládání manifestu by bylo lepší použít `ManifestManagera`
https://github.com/keboola/php-component/blob/master/src/Manifest/ManifestManager.php

Reply:
nemusíš alespoň tady zjišťovat cestu kam se to má uložit

Reply:
jj, neviem preco som ho nepouzil ...

Reply:
https://github.com/keboola/ex-azure-storage-table/pull/2/commits/fa4ad7bd358c8d003e8063c7e4dce5f498afd5f3